src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/Microsoft.Extensions.AI.Evaluation.Reporting.csproj

```diff
@@ -0,0 +1,91 @@
+    <EnablePackageValidation>false</EnablePackageValidation>
+    <MinCodeCoverage>88</MinCodeCoverage>
+    <MinMutationScore>0</MinMutationScore>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\Microsoft.Extensions.AI\Microsoft.Extensions.AI.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Extensions.AI.Evaluation\Microsoft.Extensions.AI.Evaluation.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="..\TypeScript\html-report\dist\index.html" Visible="false">
+      <LogicalName>Reporting.HTML.index.html</LogicalName>
+    </EmbeddedResource>
+  </ItemGroup>
+
+  <ItemGroup>
+    <InternalsVisibleToTest Include="Microsoft.Extensions.AI.Evaluation.Integration.Tests" />
+    <InternalsVisibleToTest Include="Microsoft.Extensions.AI.Evaluation.Reporting.Tests" />
+  </ItemGroup>
+
+  <!-- Build the report code before this, so it can be embedded.
+  Setting the BeforeTargets=DispatchToInnerBuilds allows it to build only once before
+  the multiple parallel builds are invoked for each target framework. -->
+  <Target Name="BuildTypescriptBundle"
+          BeforeTargets="DispatchToInnerBuilds"
+          Outputs="..\TypeScript\html-report\dist\index.html">
+    <Message Text="Build the Typescript bundle" Importance="high" />
+    <Exec Command="npm install" WorkingDirectory="..\TypeScript" />
+    <Exec Command="node build-if-out-of-date.js" WorkingDirectory="..\TypeScript" />
+  </Target>
+
+  <!-- Generate a version constant that can be accessed from the C# code. -->
+  <Target Name="GenerateVersionConstant" BeforeTargets="CoreCompile">
+    <PropertyGroup>
+      <ConstantsFilePath>$(IntermediateOutputPath)Constants.g.cs</ConstantsFilePath>
+      <Lines>
+// <auto-generated>
+// This file is auto-generated by MSBuild.
+// </auto-generated>
+
+namespace Microsoft.Extensions.AI.Evaluation.Reporting%3B
+
+internal static class Constants
+{
+    public const string Version = "$(Version)"%3B
+}
+      </Lines>
+    </PropertyGroup>
+
+    <Message Text="Generating $(ConstantsFilePath)." Importance="high" />
+    
+    <WriteLinesToFile File="$(ConstantsFilePath)"
+                      Lines="$(Lines)"
+                      Overwrite="true"
+                      WriteOnlyWhenDifferent="true" />
+
+    <ItemGroup>
+      <Compile Include="$(ConstantsFilePath)" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Generate a version file that can be accessed from the script that builds the Azure DevOps Extension pacakge. -->
+  <Target Name="StampVSIXPackageVersion" BeforeTargets="CoreCompile">
+    <PropertyGroup>
+      <VSIXPackageVersionFile>..\TypeScript\azure-devops-report\VSIXPackageVersion.json</VSIXPackageVersionFile>
+      <VSIXVersion>$(VersionPrefix).42424242</VSIXVersion>
+      <VSIXVersion Condition=" '$(VersionSuffixDateStamp)'!='' ">$(VersionPrefix).$(VersionSuffixDateStamp)$(VersionSuffixBuildOfTheDayPadded)</VSIXVersion>
+      <Lines>{"PackageVersion":"$(VSIXVersion)"}</Lines>
```

Comment:
Some of these properties clash with the Arcade: https://github.com/dotnet/arcade/blob/caa0982cc4f4a9c744f33336d899e6fdf310640d/src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.targets#L4-L11

If you can't use the Arcade, consider renaming your properties. A common practice is to prefix properties with the underscore to indicate the "private" nature of the property.

Reply:
Tagging @peterwald. Resolving for now but we should handle this in the follow up change you are planning for the VSIX build.